src/inmanta/agent/in_process_executor.py

```diff
@@ -572,7 +571,7 @@ async def _install(self, blueprint: executor.ExecutorBlueprint) -> None:
             await loop.run_in_executor(
                 self.process.thread_pool,
                 self._env.install_for_config,
-                list(pkg_resources.parse_requirements(blueprint.requirements)),
+                [safe_parse_requirement(requirement_name=e) for e in blueprint.requirements],
```

Comment:
Does the install command really require canonical requires? Same comment applies in different places.

Reply:
So the motivation behind this is the following: At some point, the requirements will be checked (to see if they are already installed), we will need canonical requirements for this.
- `install_from_list` depends on `install_from_index`
- `install_from_index` depends on `Install_from_config`
- `Install_from_config` will check if the provided requirements are installed

Reply:
Ok, it's unfortunate that we have to lift it to the user interface, but it's probably the lesser of two evils.